File: CS/ViewModel.cs
using System;
using System.Collections.ObjectModel;

namespace Default_MVVM {
    public class TestDataViewModel {
        public ObservableCollection<TestObjectViewModel> Records { get; set; }
        public TestDataViewModel() {
            Records = new ObservableCollection<TestObjectViewModel>();            
            for (int i = 0; i < 200; i++)
                Records.Add(new TestObjectViewModel() {
                    Text = "Row" + i,
                    Number = i,
                    Date = DateTime.Now.AddDays(i).AddHours(i)
                });
        }        
    }

    public class TestObjectViewModel {
        public string Text { get; set; }
        public int Number { get; set; }
        public DateTime Date { get; set; }
    }
}

File: VB/ViewModel.vb
Imports System
Imports System.Collections.ObjectModel

Namespace Default_MVVM
    Public Class TestDataViewModel
        Public Property Records() As ObservableCollection(Of TestObjectViewModel)
        Public Sub New()
            Records = New ObservableCollection(Of TestObjectViewModel)()
            For i As Integer = 0 To 199
                Records.Add(New TestObjectViewModel() With {.Text = "Row" & i, .Number = i, .Date = Date.Now.AddDays(i).AddHours(i)})
            Next i
        End Sub
    End Class

    Public Class TestObjectViewModel
        Public Property Text() As String
        Public Property Number() As Integer
        Public Property [Date]() As Date
    End Class
End Namespace
